### PR TITLE
[RESTEASY-1838]: Remove the SseEventSource.open() 30 seconds timeout to be fully synch

### PR DESCRIPTION
**Pull request by NicoNes**
Please read the discussion on #1456 on the reasoning that led to this PR

### CODE REVIEW

**Comment by marekkopecky** [comment]
@asoldano AFAICS, there is no way to set disableKeepAlive variable to true, so it is save to remove it, right?

File: resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceImpl.java [hunk 1]
```diff
@@ -208,12 +199,6 @@ public boolean isOpen()
       return state.get() == State.OPEN;
    }
 
-   @Override
```
Comment (marekkopecky) [inline resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceImpl.java]: If this method is removed, default jax-rs close method is used, right? Default method from spec uses 5s timeout:
```java
    @Override
    default void close() {
        close(5, TimeUnit.SECONDS);
    }
```

**Comment by marekkopecky** [comment]
If resteasy doesn't use any timeout, we need to rely on HTTP timeout, or server timeout, right? Wouldn't be better (for example) to allow to parametrize this timeout? I can imagine that some users would like to have some timeout on client. And (for example) if client doesn't return any event, users can try to send new request again. @asoldano, @jimma  WDYT?

File: resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceImpl.java (hunk 1 above)
Comment (NicoNes) [reply resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceImpl.java]: Yes, it's right

**Comment by NicoNes** [comment]
Hi @marekkopecky,

> If resteasy doesn't use any timeout, we need to rely on HTTP timeout, or server timeout, right?

Right

>  Wouldn't be better (for example) to allow to parametrize this timeout?

IMO, users can still set either `javax.ws.rs.client.ClientBuilder.connectTimeout(...)` or `javax.ws.rs.client.ClientBuilder.readTimeout(...)` to configure the `javax.ws.rs.client.Client` instance used by the SSE client to set up the timeout the way they want.

**Pull request merged**